src/services/crawl-tx/aura.registry.ts

```diff
@@ -154,4 +157,26 @@ export default class AuraRegistry {
       this.registry.register(type, _.get(this, type.slice(1)))
     );
   }
+
+  public setCosmosSdkVersionByString(version: string) {
+    this.cosmosSdkVersion = new SemVer(version);
+  }
+
+  // use for decode event attribute key value for each version cosmos sdk
+  public decodeAttributeByCosmosSdkVersion(input: string) {
```

Comment:
em viết 1 hàm trả về cái function thì mỗi lần chạy ko cần phải qua cái câu compare version
```
if (compare..) {
  this.decodeAttribute = (input) => {
     ....
  }
} else {
  ...
}

Reply:
cái này tách hàm em thấy không hay lắm, để như này thì nếu sau lại có version mới thì còn if else như này luôn
`
if (version < versionA) {}
else if (version < versionB) {}
else if (version < versionC) {}
...
`

Reply:
cái này để gen ra cái function, sau đó chỗ gọi thì cứ gọi cái function thôi
còn if else hay gì thì ko quan trọng

Reply:
thêm vào đoạn này này 
```
  public setCosmosSdkVersionByString(version: string) {
    this.cosmosSdkVersion = new SemVer(version);
  }
```

Reply:
à ý anh là kiểu này ấy hả
`public setCosmosSdkVersionByString(version: string) {
    this.cosmosSdkVersion = new SemVer(version);

    if (this.cosmosSdkVersion.compare('v0.45.99') === -1) {
      this.decodeAttributeByCosmosSdkVersion = (input) =>
        fromUtf8(fromBase64(input));
    } else {
      this.decodeAttributeByCosmosSdkVersion = (input) => input;
    }
  }`

Reply:
còn cả hàm encode nữa

Reply:
fixed